src/pages/index.js

```diff
@@ -59,55 +35,175 @@ function renderCard(cardData) {
-  renderCard(data);
-  addCardPopup.close();
+  addCardPopup.setLoading(true);
+  api
+    .addCard(data)
+    .then((res) => {
+      renderCard(res);
+      addCardPopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => addCardPopup.setLoading(false, "Create"));
 }
 
+addCardOpenButton.addEventListener("click", () => {
+  formValidators["edit-profile-form"].resetValidation();
+  addCardPopup.open();
+});
+
+// EDIT PROFILE //
+const editProfilePopup = new PopupWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+editProfilePopup.setEventListeners();
+
 function handleProfileEditSubmit(data) {
-  newUserInfo.setUserInfo(data);
-  console.log(data);
-  editProfilePopup.close();
+  editProfilePopup.setLoading(true);
+  api
+    .updateUserInfo(data.name, data.about)
+    .then((res) => {
+      newUserInfo.setUserInfo(res);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => editProfilePopup.setLoading(false, "Save"));
 }
 
-// Event Listeners //
-
 profileEditOpenButton.addEventListener("click", () => {
   const data = newUserInfo.getUserInfo();
   profileTitleInput.value = data.name;
-  profileDescriptionInput.value = data.description;
+  profileDescriptionInput.value = data.about;
   editProfilePopup.open();
 });
 
-addCardOpenButton.addEventListener("click", () => {
-  addCardFormValidator.resetValidation();
-  addCardPopup.open();
+// CHANGE AVATAR //
+const updateAvatarForm = new PopupWithForm(
+  "#avatar-image-modal",
+  handleAvatarFormSubmit
+);
+updateAvatarForm.setEventListeners();
+
+function handleAvatarFormSubmit(data) {
+  updateAvatarForm.setLoading(true);
+  api
+    .updateAvatar(data.avatar)
+    .then((res) => {
+      newUserInfo.setUserAvatar(res.avatar);
+      updateAvatarForm.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      updateAvatarForm.setLoading(false, "Save");
+    });
+}
+
+editAvatarOpenButton.addEventListener("click", () => {
+  formValidators["edit-avatar-form"].resetValidation();
+  updateAvatarForm.open();
 });
+
+// IMAGE PREVIEW //
+const imagePreview = new PopupWithImage("#image-modal");
+imagePreview.setEventListeners();
+
+// LIKE CARD //
+function handleLikeClick(item) {
+  if (!item.isLiked) {
+    api
+      .likeCard(item.getId())
+      .then((res) => {
+        item.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .unlikeCard(item.getId())
+      .then((res) => {
+        item.updateLikeStatus(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+// DELETE CARD //
+const confirmDelete = new PopupWithConfirmation("#delete-card-modal");
+confirmDelete.setEventListeners();
+
+function handleDeleteClick(card) {
+  confirmDelete.open();
+  confirmDelete.setSubmitAction(() => {
+    api
```

Comment:
Needs correction: don't forget to notify the user about the loading process by changing the submit button text

Reply:
fixed